doc/structs.asciidoc

```diff
@@ -102,6 +102,57 @@ function main = |args| {
 }
 ----
 
+=== Custom factories
+
+Golo generates two factories for structures. One for the initial values of each member and
+a second one with no parameters:
+
+[source,golo]
+----
+module test
+
+struct Point = { x, y }
+
+function main = |args| {
+  println(Point(1, 2))
+  println(Point())
+}
+----
+
+running it prints the following console output:
+
+[source,golo]
+----
+struct Point{x=1, y=2}
+struct Point{x=null, y=null}
+----
+
+NOTE: By default the no-argument factory sets every member to `null`
+
+The Factories generated by Golo can be overloaded by custom ones:
+
+[source,golo]
+----
+module test
+
+struct Point = { x, y }
+
+function Point = -> test.types.Point(0,0)
```

Comment:
Maybe you could add a note on how to call the original constructor (the `<module>.types.<struct>` thing) and how to override the immutable constructor (see @danielpetisme question in the PR comments).